File: backend-go/internal/models/document.go
package models

import "time"

type Document struct {
	ID             uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID         uint      `gorm:"index;not null" json:"user_id"`
	Title          string    `gorm:"size:256;not null" json:"title"`
	FileName       string    `gorm:"size:256;not null" json:"file_name"`
	FileType       string    `gorm:"size:16;not null" json:"file_type"`
	MimeType       string    `gorm:"size:64;default:''" json:"mime_type"`
	FileSize       int64     `gorm:"default:0" json:"file_size"`
	StoragePath    string    `gorm:"size:512;default:''" json:"storage_path"`
	ParsedStatus   string    `gorm:"size:16;default:'pending'" json:"parsed_status"`
	ParsedContent  string    `gorm:"type:text" json:"parsed_content,omitempty"`
	PageCount      int       `gorm:"default:0" json:"page_count,omitempty"`
	WordCount      int       `gorm:"default:0" json:"word_count,omitempty"`
	ReadProgress   float64   `gorm:"default:0" json:"read_progress"`
	DeletedAt      *time.Time `gorm:"index" json:"-"`
	CreatedAt      time.Time  `json:"created_at"`
	UpdatedAt      time.Time  `json:"updated_at"`
}

File: backend/app/models/document.py
import datetime

from sqlalchemy import (
    Column, Integer, String, BigInteger, Text, DateTime,
    ForeignKey, Float, Index, func,
)
from sqlalchemy.orm import relationship

from app.db.base import Base


class Document(Base):
    __tablename__ = "documents"

    id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    title = Column(String(256), nullable=False)
    file_name = Column(String(256), nullable=False)
    file_type = Column(String(16), nullable=False)  # pdf, md, docx, txt
    mime_type = Column(String(128), default="")
    file_size = Column(BigInteger, default=0)
    storage_path = Column(String(512), default="")
    preview_url = Column(String(512), default="")
    parsed_status = Column(String(16), default="pending")  # pending | done | failed
    parsed_content = Column(Text, default="")
    page_count = Column(Integer, default=0)
    word_count = Column(Integer, default=0)
    last_read_position = Column(String(256), default="")
    last_read_at = Column(DateTime, nullable=True)
    deleted_at = Column(DateTime, nullable=True)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    owner = relationship("User", backref="documents")
